google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java

```diff
@@ -1137,6 +1125,37 @@ public void close() throws Exception {
     return optionMap;
   }
 
+  static Map<Option.OptionType, ?> updateFilter(final Map<Option.OptionType, ?> options) {
+    String existingFilter = EntryListOption.OptionType.FILTER.get(options);
+    String newFilter = generateFilter(existingFilter);
+    if (newFilter.equals(existingFilter)) {
+      return options;
+    }
+    Map<Option.OptionType, Object> optionMap = Maps.newHashMap();
+    for (Map.Entry<Option.OptionType, ?> entry : options.entrySet()) {
+      optionMap.put(entry.getKey(), entry.getValue());
+    }
```

Comment:
nit: is there a reason to avoid using [this constructor](https://guava.dev/releases/19.0/api/docs/com/google/common/collect/Maps.html#newHashMap(java.util.Map))?
 `Map<Option.OptionType, Object> optionMap = Maps.newHashMap(options);`

Also the variable name optionMap vs options is a bit confusing

Reply:
Oh, great comment - thank you! Changed the code to use the proper constructor and also variable name.